From 45f7da6d5badf46439ec3e13d2c12bca4483bf93 Mon Sep 17 00:00:00 2001
From: Alex Stewart <alex.stewart@ni.com>
Date: Wed, 11 Sep 2019 13:16:11 -0500
Subject: [PATCH] libelf: Mark both fsize and msize with const attribute.

Adapted from mark Wielaard's upstream patch:
a01938d584b91e747167bb4b3f30ec300c4d6e43

```
From: Mark Wielaard <mark@klomp.org>
Date: Fri, 16 Nov 2018 09:27:00 +0100
Subject: [PATCH] libelf: Mark both fsize and msize with const attribute.

GCC9 -Wmissing-attributes pointed out that although we alias the fsize
and msize functions only fsize was marked as const. Fix by also marking
the msize definition as const.

https://sourceware.org/bugzilla/show_bug.cgi?id=23884

Signed-off-by: Mark Wielaard <mark@klomp.org>
```

Upstream-Status: Accepted
Signed-off-by: Alex Stewart <alex.stewart@ni.com>
---
 libelf/libelfP.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libelf/libelfP.h b/libelf/libelfP.h
index 7ee6625..4007301 100644
--- a/libelf/libelfP.h
+++ b/libelf/libelfP.h
@@ -463,9 +463,9 @@ extern Elf_Type __libelf_data_type (Elf *elf, int sh_type) internal_function;
    These functions cannot be marked internal since they are aliases
    of the export elfXX_fsize functions.*/
 extern size_t __elf32_msize (Elf_Type __type, size_t __count,
-			     unsigned int __version);
+			     unsigned int __version) __const_attribute__;
 extern size_t __elf64_msize (Elf_Type __type, size_t __count,
-			     unsigned int __version);
+			     unsigned int __version) __const_attribute__;
 
 
 /* Create Elf descriptor from memory image.  */
-- 
2.23.0

